apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/ApolloClient.kt

```diff
@@ -0,0 +1,41 @@
+package com.apollographql.apollo
+
+import com.apollographql.apollo.api.ApolloExperimental
+import com.apollographql.apollo.api.ExecutionContext
+import com.apollographql.apollo.api.Mutation
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.Query
+import com.apollographql.apollo.api.ScalarTypeAdapters
+import com.apollographql.apollo.executor.NetworkExecutor
+import com.apollographql.apollo.executor.RequestExecutor
+import com.apollographql.apollo.internal.RealApolloCall
+import com.apollographql.apollo.network.NetworkTransport
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+
+@ApolloExperimental
+@ExperimentalCoroutinesApi
+class ApolloClient constructor(
+    private val networkTransport: NetworkTransport,
+    private val scalarTypeAdapters: ScalarTypeAdapters = ScalarTypeAdapters.DEFAULT,
+    private val executors: List<RequestExecutor> = emptyList(),
```

Comment:
why not use the `interceptors` wording here instead ? `Executors` always remind me of https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executor.html but maybe I've done too much java.

Reply:
Sure we can name it as interceptors, no objection.